### ISSUE TITLE
Logic of petab.visualize.helper_functions.get_vis_spec_dependent_columns_dict

### ISSUE THREAD

**Issue by FFroehlich** [PEtab-dev/libpetab-python#34]
So I am currently running into the following issue:

```
Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/pydevd.py", line 1434, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "pretrain_per_sample.py", line 70, in <module>
    petab.visualize.plot_data_and_simulation(
  File "/Users/ffroehlich/Documents/HMS/mechanismEncoder/venv/lib/python3.8/site-packages/petab/visualize/plot_data_and_simulation.py", line 128, in plot_data_and_simulation
    exp_data, vis_spec = create_or_update_vis_spec(exp_data,
  File "/Users/ffroehlich/Documents/HMS/mechanismEncoder/venv/lib/python3.8/site-packages/petab/visualize/helper_functions.py", line 572, in create_or_update_vis_spec
    vis_spec = expand_vis_spec_settings(vis_spec, columns_dict)
  File "/Users/ffroehlich/Documents/HMS/mechanismEncoder/venv/lib/python3.8/site-packages/petab/visualize/helper_functions.py", line 487, in expand_vis_spec_settings
    vis_spec[select_conditions].loc[:, column].values[0])
IndexError: index 0 is out of bounds for axis 0 with size 0
```

I explicitly specify `PLOT_ID` but no `DATASET_ID` in my visualization table (if I do specify `DATASET_ID`, which is not specified in my measurements_df, petab will just show empty plots, ... bummer).

This means that I end up with https://github.com/PEtab-dev/PEtab/blob/c3d51d6ce98a533c686243dd9ba163276785fc44/petab/visualize/helper_functions.py#L558, which makes no effort at all to respect the previously defined `plot_id_list`, but instead creates a new set of plot ids of the form `plot{i}`. Those `PLOT_ID` of course don't match what's in my `vis_spec`, which means that the `select_conditions` in https://github.com/PEtab-dev/PEtab/blob/c3d51d6ce98a533c686243dd9ba163276785fc44/petab/visualize/helper_functions.py#L484 will never match anything.

I am at loss how this is conceptually supposed to work at all, so I would appreciate if anybody could enlighten me such that I can fix this issue.

**Comment by plakrisenko** [comment]
I think this issue might be the same as #28. Should be fixed now.

**Comment by plakrisenko** [comment]
> [dream_cytof__EGFR_MAPK_AKT_STAT__c184A1__visualization.txt](https://github.com/PEtab-dev/PEtab/files/5934172/dream_cytof__EGFR_MAPK_AKT_STAT__c184A1__visualization.txt)
> 
> I have attached one of the visualization tables I used. The data consists of a couple of dynamic observables that are measured in multiple conditions, where I would like individual plots for every obervables with line plots for every observable, as well as a couple of static observables that are only measured once in the baseline condition, where I would like a bar plot of all those observables in a single plot. I tried using `datasetId` for grouping, but I ran into issues since I am not using (and probably shouldn't since the grouping is different from what I want in the plots) `datasetId` in the measurement table.

This visualization specification should work now (there indeed was a problem related to plotIds, see #28). 
However, with such visualization table you will be plotting the same plots multiple times. E.g. lines 2, 16, 30 and 44 are the same in terms of **what** to plot. The second line already will result in multiple lines corresponding to different conditions for observable `pEIF4EBP1_T37_T46_obs`. All these lines will have label `EGF`. `LegendEntry` column doesn't affect what should be plotted.

**Comment by dweindl** [comment]
Fixed in `v0.1.19`
